### ISSUE TITLE
/p home does not work properly

### ISSUE THREAD

**Issue by JaTiTV** [IntellectualSites/PlotSquared#3565]
### Server Implementation

Paper

### Server Version

1.18.2

### Describe the bug

When I put my home in my house or in my basement I am always teleported to the top when I use '/p h'.
With '/p visit <player>' is the same problem

### To Reproduce

1. /p sethome under a block
2. /p home
you will be teleported to the top block

### Expected behaviour

Player should spawn in correct location

### Plot Debugpaste

https://athion.net/ISPaster/paste/view/8365bc4a1ca545e38b2c9f531a0dfaf9

### PlotSquared Version

6.6.2-Premium

### Checklist

- [X] I have included a Plot debugpaste.
- [X] I am using the newest build from https://www.spigotmc.org/resources/77506/ and the issue still persists.

**Comment by osfanbuff63** [comment]
#3536

**Comment by JaTiTV** [comment]
This is not the same!

**Comment by Myszku** [comment]
> This is not the same!

Yesterday I reported the same issue, but I figured out where is the problem and I think it was my fault but idk. This is what it look like: https://imgur.com/a/C31Jzqz
In worlds.yml I set min_gen_height to 0, previously I had -1 there. On my server I have disabled building under Y0, so these works for me.

This is my world section in worlds.yml file:
`    world:
      max_gen_height: 255
      max_height: 256
      gamemode: creative
      min_gen_height: 0
      min_height: 2`
